pkg/threatest/matchers/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,137 @@
+package elasticsearch
+
+import (
+	"fmt"
+	"strings"
+	log "github.com/sirupsen/logrus"
+	"encoding/json"
+	"errors"
+)
+
+func FilterByUuidPresence(alerts []ElasticsearchQueryHit, uuid string) []ElasticsearchQueryHit {
+	var filteredAlerts []ElasticsearchQueryHit
+	var containsUuid bool
+	for _,alert := range alerts {
+		containsUuid = false
+		for _,v := range alert.Source {
+			if strings.Contains(v.(string), uuid) {
+				containsUuid = true
+				break
+			}
+		}
+		if containsUuid {
+			filteredAlerts = append(filteredAlerts, alert)
+		}
+	}
+	return filteredAlerts
+}
+
+func StripHTTPStatusCode(response string) (string, error) {
+    index := strings.Index(response, "{")
+    if index != -1 {
+        return response[index:], nil
+    } else {
+        return "", errors.New("No '{' found in Elasticsearch query response")
+    }
+}
+
+func RetrieveAlerts(m *ElasticsearchAlertGeneratedAssertion, uuidField, ruleName string) ([]ElasticsearchQueryHit, error) {
+	// The alias for the Elasticsearch index where alerts are stored
+	const ALERT_INDEX string = ".siem-signals-default"
+	// Construct the query necessary to find the alert
+	query := `
+	{
+		"_source": [ "%s" ],
+		"query": {
+			"bool": { 
+				"filter": [ 
+					{ "range": { "@timestamp": { "gte": "now-3d" }}},
+					{ "term":  { "kibana.alert.rule.name": "%s" }},
+					{ "term":  { "kibana.alert.workflow_status": "open" }}
+				]
+			}
+		}
+	}`
+	// Template in the field we expect to find the UUID in, and the rule we hope was triggered.
+	query = fmt.Sprintf(query, uuidField, ruleName)
+	// Query the Elasticsearch API
+	res, err := m.AlertAPI.Search(
+		m.AlertAPI.Search.WithIndex(ALERT_INDEX),
+		m.AlertAPI.Search.WithBody(strings.NewReader(query)),
+	)
+	if err != nil {
+		log.Fatal("Error while running Elasticsearch query")
+		return nil, err
+	}
+	// Parse the response
+	strippedResponse, err := StripHTTPStatusCode(res.String())
+	if err != nil {
+		log.Fatal("Error while stripping prepended HTTP status code")
```

Comment:
Do you want to include the error here? `Fatal` will exit immediately and so the return won't do anything and you will lose the original error. Although maybe you don't need it because there is only one error it can be :thinking: